sympy/discrete/convolutions.py

```diff
@@ -88,6 +91,11 @@ def convolution(a, b, cycle=0, dps=None, prime=None, dyadic=None, subset=None):
         ls = convolution_fwht(a, b)
     elif subset:
         ls = convolution_subset(a, b)
+    elif all(isinstance(e, (int, Rational)) for e in chain(a, b)):
```

Comment:
Should we make it more granular? For around 500 numbers in range [1,1000] it is 2-3 times faster to do so.
```python
def do(a):
    ints = []
    dens = []
    for i in a:
        if isinstance(i, int):
            ints.append(i)
        elif isinstance(i, Integer):
            ints.append(int(i))
        elif isinstance(i, Rational):
            ints.append(i.p)
            dens.append(i.q)
        else:
            return
    if not dens:
        return ints, 1
    LCM = ilcm(*dens)
    return [i*LCM for i in ints], LCM
ls = None
da = do(a)
if da is not None:
    db = do(b)
    if db is not None:
        ia, ma = da
        ib, mb = db
        raw = convolution_int(ia, ib)
        den = ma*mb
        if den == 1:
            ls = [Integer(i) for i in convolution_int(ia, ib)]
        else:
            ls = [Rational(i, den) for i in convolution_int(ia, ib)]
if ls is None:
    ls = convolution_fft(a, b, dps)
```

Reply:
:+1: I've had to modify this slightly (we need to multiply the whole number by the lcm not just the numerator in `return [i*LCM for i in ints]`), but I've found it to still be a fair amount faster (I've added you as a co-auth, I hope you don't mind). 

As an aside, do you think it would be worth adding a benchmark for some of these cases in https://github.com/sympy/sympy_benchmarks?

Reply:
I am not sure what the selection criteria for inclusion in the benchmark suite is. Perhaps @asmeurer or @oscarbenjamin could comment.

Reply:
I think it's fine to add a benchmark. Ideally we will have benchmarks for all the various parts of SymPy. If we truly had benchmarks for *everything* we would need to split them up somehow, but we aren't nearly at that point yet.